### PR TITLE
Add gaussian smoothing

### PR DESCRIPTION
**Pull request by Sbozzolo**
This PR introduces our gaussian smoothing function and removes the tower of dependencies that comes with `ImageFiltering`. Removes lots of dependencies and reduces time to `using ClimaAtmos` by 10 %.

Our gaussian smoothing function is not precisely identical to `imfilter` (e.g., how we treat window sizes and edges), but results are nearly identical. 

image

Same, but with `imfilter

image

Closes https://github.com/CliMA/ClimaAtmos.jl/issues/2323

Before (on 1.10)
```
julia> @time using ClimaAtmos
  4.011574 seconds (5.14 M allocations: 453.871 MiB, 12.37% gc time, 8.76% compilation time: 75% of which was recompilation)
```
After (on 1.10)
```
julia> @time using ClimaAtmos
  3.619062 seconds (4.67 M allocations: 420.374 MiB, 13.62% gc time, 9.51% compilation time: 75% of which was recompilation)
```

### CODE REVIEW

File: test/orographic_gravity_wave/ogwd_3d.jl [hunk 1]
```diff
@@ -81,7 +80,7 @@ earth_spline = NCDataset(data_path) do data
     lat = Array(data["latitude"])
     # Apply Smoothing
     smooth_degree = 15
-    esmth = imfilter(zlevels, Kernel.gaussian(smooth_degree))
+    esmth = gaussian_smooth(zleleves, smooth_degree)
```
Comment (akshaysridhar) [inline test/orographic_gravity_wave/ogwd_3d.jl:83]:
```suggestion
    esmth = gaussian_smooth(zlevels, smooth_degree)
```

File: src/solver/type_getters.jl [hunk 2]
```diff
@@ -143,7 +142,7 @@ function get_spaces(parsed_args, params, comms_ctx)
             lat = Array(data["latitude"])
             # Apply Smoothing
             smooth_degree = Int(parsed_args["smoothing_order"])
-            esmth = imfilter(zlevels, Kernel.gaussian(smooth_degree))
+            esmth = gaussian_smooth(smooth_degree)
```
Comment (akshaysridhar) [inline src/solver/type_getters.jl:145]: Are we missing the array as an `arg` here?

Comment (akshaysridhar) [inline src/solver/type_getters.jl:145]:
```suggestion
            esmth = gaussian_smooth(zlevels, smooth_degree)
```

Comment (Sbozzolo) [reply src/solver/type_getters.jl:145]: Thanks!

Comment (Sbozzolo) [reply src/solver/type_getters.jl]: Yes we are

**Comment by akshaysridhar** [comment]
LGTM

File: src/solver/type_getters.jl [hunk 3]
```diff
@@ -143,7 +142,7 @@ function get_spaces(parsed_args, params, comms_ctx)
             lat = Array(data["latitude"])
             # Apply Smoothing
             smooth_degree = Int(parsed_args["smoothing_order"])
-            esmth = imfilter(zlevels, Kernel.gaussian(smooth_degree))
+            esmth = gaussian_smooth(zlevels, smooth_degree)
```
Comment (akshaysridhar) [inline src/solver/type_getters.jl:145]: `CA.gaussian_smooth` ?

**Pull request merged**